### PR TITLE
[Merged by Bors] - chore: rename `library_search`/`propose`/`rewrites` to `apply?`/`have?`/`rw?`

### PR DESCRIPTION
**Pull request by kim-em**
Also adds `exact?` which is like `apply?` but fails if it can't close the goal.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alexjbest** [comment]
I think a lot of people have `library_search` muscle memory at this point, if we are going to merge this can we at least add a `library_search` alias for the time being (with a big deprecation warning perhaps)

File: test/LibrarySearch/basic.lean [hunk 1]
```diff
@@ -129,9 +129,9 @@ theorem Bool_eq_iff {A B: Bool} : (A = B) = (A ↔ B) :=
...
 
 theorem Bool_eq_iff2 {A B: Bool} : (A = B) = (A ↔ B) :=
-  by library_search -- exact Bool_eq_iff
+  by apply? -- exact Bool_eq_iff
 
 assert_no_sorry Bool_eq_iff2
 
--- Example from https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/library_search.20regression/near/354025788
-example {r : α → α → Prop} : Function.Surjective (Quot.mk r) := by library_search
+-- Example from https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/apply?.20regression/near/354025788
```
Comment (alexjbest) [inline test/LibrarySearch/basic.lean:136]: Find and replace too aggressive here

Comment (semorrison) [inline test/LibrarySearch/basic.lean:136]:
```suggestion
-- Example from https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/library_search.20regression/near/354025788
```

**Comment by fpvandoorn** [comment]
bors merge

File: Mathlib/Tactic/LibrarySearch.lean [hunk 2]
```diff
@@ -15,8 +15,9 @@ import Mathlib.Control.Basic
 /-!
 # Library search
 
-This file defines a tactic `library_search`
-and a term elaborator `library_search%`
+This file defines tactics `exact?` and `apply?`,
+(formerly known as `library_search`)
```
Comment (eric-wieser) [inline Mathlib/Tactic/LibrarySearch.lean:19]: Should we rename the import?

**Pull request closed without merging**